docs/reference/stdlib.md

```diff
@@ -0,0 +1,693 @@
+// integer value (milliseconds)
+oneSecond = Duration.of(1000)
+
+// simple string value
+oneHour = Duration.of('1h')
+
+// complex string value
+complexDuration = Duration.of('1day 6hours 3minutes 30seconds')
+```
+
+Durations can be compared like numbers, and they support basic arithmetic operations:
+
+```groovy
+a = 1.h
+b = 2.h
+
+assert a < b
+assert a + a == b
+assert b - a == a
+assert a * 2 == b
+assert b / 2 == a
+```
+
+The following methods are available for a `Duration` object:
+
+`getDays()`, `toDays()`
+: Get the duration value in days (rounded down).
+
+`getHours()`, `toHours()`
+: Get the duration value in hours (rounded down).
+
+`getMillis()`, `toMillis()`
+: Get the duration value in milliseconds.
+
+`getMinutes()`, `toMinutes()`
+: Get the duration value in minutes (rounded down).
+
+`getSeconds()`, `toSeconds()`
+: Get the duration value in seconds (rounded down).
+
+(stdlib-classes-memoryunit)=
+
+### MemoryUnit
+
+A `MemoryUnit` represents a quantity of bytes.
+
+You can create a memory unit by adding a unit suffix to an integer, e.g. `1.GB`. The following suffixes are available:
+
+| Unit | Description |
+| ---- | ----------- |
+| `B`  | Bytes       |
+| `KB` | Kilobytes   |
+| `MB` | Megabytes   |
+| `GB` | Gigabytes   |
+| `TB` | Terabytes   |
+| `PB` | Petabytes   |
+| `EB` | Exabytes    |
+| `ZB` | Zettabytes  |
+
+:::{note}
+Technically speaking, a kilobyte is equal to 1000 bytes, whereas 1024 bytes is called a "kibibyte" and abbreviated as "KiB", and so on for the other units. In practice, however, kilobyte is commonly understood to mean 1024 bytes, and Nextflow follows this convention in its implementation as well as this documentation.
+:::
+
+You can also create a memory unit with `MemoryUnit.of()`:
+
+```groovy
+// integer value (bytes)
+oneKilobyte = MemoryUnit.of(1024)
+
+// string value
+oneGigabyte = MemoryUnit.of('1 GB')
+```
+
+Memory units can be compared like numbers, and they support basic arithmetic operations:
+
+```groovy
+a = 1.GB
+b = 2.GB
+
+assert a < b
+assert a + a == b
+assert b - a == a
+assert a * 2 == b
+assert b / 2 == a
+```
+
+The following methods are available for a `MemoryUnit` object:
+
+`getBytes()`, `toBytes()`
+: Get the memory value in bytes (B).
+
+`getGiga()`, `toGiga()`
+: Get the memory value in gigabytes (rounded down), where 1 GB = 1024 MB.
+
+`getKilo()`, `toKilo()`
+: Get the memory value in kilobytes (rounded down), where 1 KB = 1024 B.
+
+`getMega()`, `toMega()`
+: Get the memory value in megabytes (rounded down), where 1 MB = 1024 KB.
+
+`toUnit( unit )`
+: Get the memory value in terms of a given unit (rounded down). The unit can be one of: `'B'`, `'KB'`, `'MB'`, `'GB'`, `'TB'`, `'PB'`, `'EB'`, `'ZB'`.
+
+(stdlib-classes-path)=
+
+### Path
+
+The `file()` method returns a [Path](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html), so any method defined for Path can also be used in a Nextflow script.
+
+#### Getting attributes
```

Comment:
I think we can combine constants and functions into a single section e.g. "Global variables and functions" and lift everything from "Classes" up one level. I will try that

Reply:
Updated, let me know what you think

Reply:
Nice change! I've had a quick skim again and found another 4 places with an additional level.

- Workflows > Invoking Processes > Process outputs >
- Sharing pipelines > Managing dependencies > Code >
- Azure > Microsoft Entra > Managed identities >
- Fusion file system > Getting started > AWS Cloud >

If you think they are quick to fix it would be great to update them now too, otherwise, they can wait.

Maybe there is a way to reformat the sidebar so anything at a certain subheading level isn't listed. I'll look into this more tomorrow.

Reply:
These cases are trickier and mostly out of scope, so I'll leave them for a future iteration. Some of those pages will likely be refactored or moved anyway.